src/progress/progress_set.rs

```diff
@@ -401,12 +425,15 @@ impl ProgressSet {
             ));
         }
 
-        self.configuration.learners.remove(&id);
-        self.configuration.voters.remove(&id);
-        let removed = self.progress.remove(&id);
-
-        self.assert_progress_and_configuration_consistent();
-        Ok(removed)
+        if self.configuration.voters.remove(&id) {
+            self.assert_progress_and_configuration_consistent();
+            return Ok(());
+        }
+        if self.configuration.learners.remove(&id) {
+            self.assert_progress_and_configuration_consistent();
+            return Ok(());
+        }
+        Err(Error::NotExists(id, "configuration"))
```

Comment:
Is it safe to return Error? `Ok` is returned in the pass.

Reply:
It depends on how to deal with the `Error` outside. It's ignored in the caller, so it's safe to return `Error` here. Handle `Error` outside is just for unifying logs.